src/validation.test.ts

```diff
@@ -17,100 +10,14 @@ import {
   validateTLSClientCert,
   validateTLSClientKey,
 } from 'validation';
+
 jest.unmock('utils');
 
 describe('trivial cases', () => {
-  it('should be a valid http check', async () => {
-    const check: Check = {
-      job: 'job-1',
-      alertSensitivity: AlertSensitivity.None,
-      target: 'https://grafana.com/',
-      frequency: 10000,
-      timeout: 1000,
-      labels: [],
-      probes: [1],
-      enabled: true,
-      basicMetricsOnly: true,
-      settings: {
-        http: {
-          compression: HTTPCompressionAlgo.none,
-          method: HttpMethod.GET,
-          ipVersion: IpVersion.V4,
-          noFollowRedirects: false,
-        },
-      },
-    };
-
-    expect(validateCheck(check)).toBe(false);
-  });
-
-  it('should be a valid ping check', async () => {
-    const check: Check = {
-      job: 'job-1',
-      alertSensitivity: AlertSensitivity.None,
-      target: 'grafana.com',
-      frequency: 10000,
-      timeout: 1000,
-      labels: [],
-      probes: [1],
-      enabled: true,
-      basicMetricsOnly: true,
-      settings: {
-        ping: {
-          ipVersion: IpVersion.V4,
-          dontFragment: true,
-        },
-      },
-    };
-
-    expect(validateCheck(check)).toBe(false);
-  });
-
-  it('should be a valid dns check', async () => {
-    const check: Check = {
-      job: 'job-1',
-      alertSensitivity: AlertSensitivity.None,
-      target: 'grafana.com',
-      frequency: 10000,
-      timeout: 1000,
-      labels: [],
-      probes: [1],
-      enabled: true,
-      basicMetricsOnly: true,
-      settings: {
-        dns: {
-          recordType: DnsRecordType.A,
-          server: 'dns.google',
-          ipVersion: IpVersion.V4,
-          protocol: DnsProtocol.TCP,
-          port: 53,
-        },
-      },
-    };
-
-    expect(validateCheck(check)).toBe(false);
-  });
-
-  it('should be a valid tcp check', () => {
-    const check: Check = {
-      job: 'job-1',
-      alertSensitivity: AlertSensitivity.None,
-      target: 'grafana.com:80',
-      frequency: 10000,
-      timeout: 1000,
-      labels: [],
-      probes: [1],
-      enabled: true,
-      basicMetricsOnly: true,
-      settings: {
-        tcp: {
-          ipVersion: IpVersion.V4,
-          tls: false,
-        },
-      },
-    };
-
-    expect(validateCheck(check)).toBe(false);
+  BASIC_CHECK_LIST.forEach((check) => {
+    test(`should validate ${Object.keys(check.settings)[0]} check`, () => {
+      expect(validateCheck(check)).toBe(false);
+    });
```

Comment:
I just want to pick on this test in particular because I think this really shows the power of having unified data and more closely mocking a real world database for our integration tests.

As all tests are now based on the tests within `BASIC_CHECK_LIST`, you should be able to update any value in those tests and no longer break any other test -- unless you do something which would be invalid, so if for any reason our type checker doesn't pick it up, then this test should as a fallback.

On that note, I have another tech debt PR planned to make the Check typing _much_ tighter.